src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs

```diff
@@ -2101,6 +2101,18 @@ private static bool IsUserDefinedTrueOrFalse(BoundUnaryOperator @operator)
                             break;
                     }
                 }
+                else if (boundNodeForSyntacticParent is BoundBinaryOperator binaryOp
+                         && binaryOp.IsNullableComparedWithNull(out TypeSymbol binaryType)
+                         && boundExpr.IsLiteralNull())
+                {
+                    // When a nullable value type is compared with null, we don't generate a BoundConversion in the tree.
+                    // Adjust the converted type to be the nullable conversion we expected.
```

Comment:
>/ Adjust the converted type to be the nullable conversion we expected. [](start = 21, length = 70)

I do not think we want to do this, this is a null check, null isn't converted to the ```Nullable<T>``` type